R/features.R

```diff
@@ -176,53 +176,95 @@ helper_feature <- function(m, feature_sample) {
 #'
 #' @param data data.table or data.frame. Includes all the features (both factors and possibly others).
 #'
+#' @param newdata data.table or data.frame. New data (features) that has the same
+#' features as \code{data}.
+#'
 #' @return A list that contains the following entries:
 #' \describe{
+#' \item{obj}{List, Contains \describe{
 #' \item{features}{Vector. Contains the names of all the features in \code{data}.}
 #' \item{factor_features}{Vector. Contains the names of all the factors in \code{data}.}
 #' \item{factor_list}{List. Contains each factor and its vector of levels.}
 #' \item{contrasts_list}{List. Contains all the contrasts of the factors.}
+#' }}
+#' \item{model.matrix}{A data.frame containing all of the factors in \code{new_data} as
+#' one-hot encoded variables.}
 #' }
 #'
+#'
 #' @export
 #'
 #' @author Annabelle Redelmeier
 #'
 #' @examples
 #'
 #' data("Boston", package = "MASS")
-#'
-#' x_var <- c("lstat", "chas", "rad", "indus")
+#' x_var <- c("lstat", "rm", "dis", "indus")
 #' y_var <- "medv"
+#' x_train <- as.data.frame(Boston[401:411, x_var])
+#' y_train <- Boston[401:408, y_var]
+#' x_test <- as.data.frame(Boston[1:4, x_var])
 #'
-#' # convert to factors
-#' Boston$rad = as.factor(Boston$rad)
-#' Boston$chas = as.factor(Boston$chas)
-#'
-#' x_train <- Boston[-1:-6, x_var]
-#' y_train <- Boston[-1:-6, y_var]
-#' x_test <- Boston[1:6, x_var]
+#' # convert to factors for illustational purpose
+#' x_train$rm <- factor(round(x_train$rm))
+#' x_test$rm <- factor(round(x_test$rm), levels = levels(x_train$rm))
 #'
-#' dummylist <- make_dummies(data = rbind(x_train, x_test))
+#' dummylist <- make_dummies(data = rbind(x_train, x_test), newdata = x_test)
 #'
-make_dummies <- function(data) {
+make_dummies <- function(data, newdata) {
```

Comment:
Suggesting data -> x_train or data_train, newdata -> x_test or data_test. There is nothing new about newdata here (while there actually is in `apply_dummies`.